File: 이것이_취업을_위한_코딩_테스트다/ch4 Implementation/KingdomKnight.java
import java.util.Scanner;

public class KingdomKnight {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String inputData = sc.nextLine();

        int row = inputData.charAt(0) - 'a' + 1;// 행: a~h->1~8
        int col = inputData.charAt(1) - '0'; // 열:1~8

        // 나이트가 이동할 수 있는 8가지 방향 정의
        int[] dx = { -2, -2, -1, -1, 2, 2, 1, 1 };
        int[] dy = { -1, 1, -2, 2, -1, 1, -2, 2 };

        int cnt = 0;

        for (int i = 0; i < 8; i++) {
            int nextRow = row + dx[i];
            int nextCol = col + dy[i];

            // 범주밖이면 다음걸로 진행
            if (nextRow < 1 || nextRow > 8 || nextCol < 1 || nextRow > 8) {
                continue;
            }
            ++cnt;
        }
        sc.close();
        System.out.println(cnt);

    }
}

File: 이것이_취업을_위한_코딩_테스트다/ch4 Implementation/kingdom_knight.py
data = input()
# ord 함수는 주어진 문자의 아스키 코드 값을 반환.
# a~h -> 1~8로 아스키 코드 후 정수형으로 고침
row = int(ord(data[0])) - int(ord("a")) + 1
col = int(data[1])  # 문자열을 정수형으로 고침.


# 나이트가 이동 가능한 좌표의 경우를 리스트로 표현
steps = [(-2, -1), (-2, 1), (-1, -2), (-1, 2), (1, -2), (1, 2), (2, -1), (2, 1)]

# 경우의 수
cnt = 0

for step in steps:
    next_row = row + step[1]
    next_col = col + step[0]
    if (next_row < 1) or (next_row > 8) or (next_col < 1) or (next_col > 8):
        continue
    else:
        cnt += 1

print(cnt)
